File: src/WebApp/ApiGateway/ApiGateway.Infrastructure/Employer/Models/Vacancy.cs
using System;

namespace ApiGateway.Infrastructure.Employer.Models
{
    public class Vacancy
    {
        public int Id { get; set; }
        public string PublishedBy { get; set; }
        public DateTime PublishedDate { get; set; }
        public int NoOfVacancies { get; set; }
        public string MinimumQualification { get; set; }
        public string JobDescription { get; set; }
        public string ExperienceRequired { get; set; }
        public string Location { get; set; }
        public DateTime LastDateToApply { get; set; }
        public int MinSalary { get; set; }

        public int MaxSalary { get; set; }

        // foreign key to employer entity config
        public int EmployerEntityId { get; set; }
        public EmployerEntity EmployerEntity { get; set; }
    }
}

File: src/Client/job-portal-client/src/app/_models/vacancy.ts
export interface Vacancy {
  id: number;
  publishedBy: string;
  publishedDate: Date;
  noOfVacancies: number;
  minimumQualification: string;
  jobDescription: string;
  experienceRequired: string;
  location: string;
  lastDateToApply: Date;
  minSalary: number;
  maxSalary: number;
  employerEntityId: number;
}

export enum ToOrderBy {
  MinSalaryAscending = 0,
  MinSalaryDescending = 1,
  MaxSalaryDescending = 2,
  LastDateToApply = 3,
  PublishedDate = 4,
}
